apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/AdjustAndDepositModal.tsx

```diff
@@ -52,35 +64,103 @@ export const AdjustAndDepositModal: FC<AdjustAndDepositModalProps> = ({
 }) => {
   const [adjustType, setAdjustType] = useState(AdjustType.Deposit);
   const [value, setValue, amount] = useWeiAmountInput('');
+  const { account } = useAccount();
+  const { onDeposit, onWithdraw } = useHandleMarketMaking(onClose);
+  const { balanceA, loadingA } = useGetUserInfo(pool);
+
+  const decimalAmount = useMemo(
+    (): Decimal => decimalic(amount.toString()),
+    [amount],
+  );
+
+  const isAmountZero = useMemo(() => amount.isZero(), [amount]);
+  const { tokenPoolBalance } = useGetPoolBalance(
+    isAmountZero,
+    adjustType,
+    loadingA,
+    pool,
+    account,
+  );
+
+  const poolWeiAmount = calculatePoolWeiAmount(
+    Decimal.fromBigNumberString(amount.toString()),
+    balanceA,
+    tokenPoolBalance,
+  );
+
+  const token = useMemo(() => pool.assetA, [pool.assetA]);
+  const isDeposit = useMemo(
+    () => adjustType === AdjustType.Deposit,
+    [adjustType],
+  );
   const [hasDisclaimerBeenChecked, setHasDisclaimerBeenChecked] =
     useState(false);
 
+  const { balance: maxTokenToDepositAmount } = useMaxAssetBalance(
+    token,
+    defaultChainId,
+    isDeposit
+      ? GAS_LIMIT.MARKET_MAKING_ADD_LIQUIDITY
+      : GAS_LIMIT.MARKET_MAKING_REMOVE_LIQUIDITY,
+  );
+
+  const maxBalance = useMemo(
+    () => (isDeposit ? maxTokenToDepositAmount : balanceA),
+    [isDeposit, balanceA, maxTokenToDepositAmount],
+  );
+
   const handleMaxClick = useCallback(
     () => setValue(maxBalance.toString()),
-    [setValue],
+    [maxBalance, setValue],
   );
 
-  const token = useMemo(() => pool.assetA, [pool.assetA]);
-
-  const decimalAmount = useMemo(
-    (): Decimal => Decimal.fromBigNumberString(amount.toString()),
-    [amount],
+  const { amount: expectedTokenAmount } = useGetExpectedTokenAmount(
+    pool,
+    decimalic(value),
+  );
+  const minReturn1 = getMinReturn(decimalAmount);
+  const minReturn2 = getMinReturn(
+    Decimal.from(toWei(expectedTokenAmount.toString()).toString()),
   );
```

Comment:
```suggestion
  const minReturn2 = getMinReturn(expectedTokenAmount);
```
`expectedTokenAmount` is already `Decimal`

Reply:
yes, `expectedTokenAmount` is already `Decimal`, but I need to make it toWei to get right value which will be sent to the contract